phpspy.c

```diff
@@ -143,7 +151,26 @@ static void dump_trace(pid_t pid, unsigned long long executor_globals_addr) {
         remote_execute_data = execute_data.prev_execute_data;
         depth += 1;
     }
-    if (wrote_trace) printf("%s", opt_trace_delim);
+    if (wrote_trace) {
+        if (opt_capture_req) {
+            try_copy_proc_mem("sapi_globals", (void*)sapi_globals_addr, &sapi_globals, sizeof(sapi_globals));
+            if (sapi_globals.request_info.request_uri) {
+                try_copy_proc_mem("uri", sapi_globals.request_info.request_uri, &uri, STR_LEN+1);
+            } else {
+                uri[0] = '-';
+                uri[1] = '\0';
+            }
+            if (sapi_globals.request_info.path_translated) {
+                try_copy_proc_mem("path", sapi_globals.request_info.path_translated, &path, STR_LEN+1);
+            } else {
+                path[0] = '-';
+                path[1] = '\0';
+            }
+            printf("# %f %s %s%s", sapi_globals.global_request_time, uri, path, opt_trace_delim);
```

Comment:
some stacks end up printed without this line. example output from my VM with wordpress:

```
21 <main> /srv/www/schwertly.com/public_html/index.php:17
# 1531446399.890827 /index.php /srv/www/schwertly.com/public_html/index.php

0 autoptimizeScripts::read /srv/www/schwertly.com/public_html/wp-content/plugins/autoptimize/classes/autoptimizeScripts.php:96
1 autoptimize_end_buffering /srv/www/schwertly.com/public_html/wp-content/plugins/autoptimize/autoptimize.php:293
2 ob_end_flush <internal>:-1
3 wp_ob_end_flush_all /srv/www/schwertly.com/public_html/wp-includes/functions.php:3743
4 WP_Hook::apply_filters /srv/www/schwertly.com/public_html/wp-includes/class-wp-hook.php:286
5 WP_Hook::do_action /srv/www/schwertly.com/public_html/wp-includes/class-wp-hook.php:310
6 do_action /srv/www/schwertly.com/public_html/wp-includes/plugin.php:453
7 shutdown_action_hook /srv/www/schwertly.com/public_html/wp-includes/load.php:679


0 <main> /srv/www/schwertly.com/public_html/wp-content/plugins/cloudflare/src/WordPress/WordPressClientAPI.php:8
```

I think it's `try_copy_proc_mem` returning when it fails to pull sapi data